xbmc/music/Album.cpp

```diff
@@ -135,14 +138,19 @@ CAlbum::CAlbum(const CFileItem& item)
         musicBrainzAlbumArtistHints = tag.GetAlbumArtist();
     }
 
+    // Try to get number of artist sort names and musicbrainz ids to match. Split sort names 
+    // further using multiple possible delimiters, over single separator applied in Tag loader
+    if (artistSort.size() != tag.GetMusicBrainzAlbumArtistID().size())
+      artistSort = StringUtils::SplitMulti(artistSort, { ";", ":", "|", "#" });
+
     for (size_t i = 0; i < tag.GetMusicBrainzAlbumArtistID().size(); i++)
```

Comment:
Since it is checking the element of one array against the element in same position in another using the array index makes more sense to me. Or is there another more elegent way to do this?